tests/components/arcam_fmj/test_config_flow.py

```diff
@@ -3,35 +3,152 @@
 import pytest
 
 from homeassistant import data_entry_flow
-from homeassistant.components.arcam_fmj.config_flow import ArcamFmjFlowHandler
-from homeassistant.components.arcam_fmj.const import DOMAIN
+from homeassistant.components import ssdp
+from homeassistant.components.arcam_fmj.config_flow import (
+    ArcamFmjFlowHandler,
+    get_entry_client,
+)
+from homeassistant.components.arcam_fmj.const import DOMAIN, DOMAIN_DATA_ENTRIES
+from homeassistant.const import CONF_HOST, CONF_PORT
 
-from .conftest import MOCK_CONFIG, MOCK_NAME
+from .conftest import (
+    MOCK_CONFIG_ENTRY,
+    MOCK_HOST,
+    MOCK_NAME,
+    MOCK_PORT,
+    MOCK_UDN,
+    MOCK_UUID,
+)
 
 from tests.common import MockConfigEntry
 
+MOCK_UPNP_DEVICE = f"""
+<root xmlns="urn:schemas-upnp-org:device-1-0">
+  <device>
+    <UDN>{MOCK_UDN}</UDN>
+  </device>
+</root>
+"""
 
-@pytest.fixture(name="config_entry")
-def config_entry_fixture():
-    """Create a mock Arcam config entry."""
-    return MockConfigEntry(domain=DOMAIN, data=MOCK_CONFIG, title=MOCK_NAME)
+MOCK_UPNP_LOCATION = f"http://{MOCK_HOST}:8080/dd.xml"
 
+MOCK_DISCOVER = {
+    ssdp.ATTR_UPNP_MANUFACTURER: "ARCAM",
+    ssdp.ATTR_UPNP_MODEL_NAME: " ",
+    ssdp.ATTR_UPNP_MODEL_NUMBER: "AVR450, AVR750",
+    ssdp.ATTR_UPNP_FRIENDLY_NAME: f"Arcam media client {MOCK_UUID}",
+    ssdp.ATTR_UPNP_SERIAL: "12343",
+    ssdp.ATTR_SSDP_LOCATION: f"http://{MOCK_HOST}:8080/dd.xml",
+    ssdp.ATTR_UPNP_UDN: MOCK_UDN,
+    ssdp.ATTR_UPNP_DEVICE_TYPE: "urn:schemas-upnp-org:device:MediaRenderer:1",
+}
 
-async def test_single_import_only(hass, config_entry):
-    """Test form is shown when host not provided."""
-    config_entry.add_to_hass(hass)
+
+@pytest.fixture(name="flow")
+def flow_fixture(hass):
+    """Create a mock flow for use in tests."""
     flow = ArcamFmjFlowHandler()
     flow.hass = hass
-    result = await flow.async_step_import(MOCK_CONFIG)
-    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
-    assert result["reason"] == "already_setup"
+    flow.context = {}
+    return flow
 
 
-async def test_import(hass):
-    """Test form is shown when host not provided."""
-    flow = ArcamFmjFlowHandler()
-    flow.hass = hass
-    result = await flow.async_step_import(MOCK_CONFIG)
+async def test_ssdp(hass, flow):
+    """Test a ssdp import flow."""
+    result = await flow.async_step_ssdp(MOCK_DISCOVER)
```

Comment:
Please use the core interface to start the test. Use `hass.config_entries.flow.async_init`. See eg the hue config flow tests.

Reply:
damn you make my life challenging ;)